virtcontainers/clh.go

```diff
@@ -373,8 +375,29 @@ func (clh *cloudHypervisor) getThreadIDs() (vcpuThreadIDs, error) {
 }
 
 func (clh *cloudHypervisor) hotplugAddDevice(devInfo interface{}, devType deviceType) (interface{}, error) {
-	clh.Logger().WithField("function", "hotplugAddDevice").Warn("hotplug add device not supported")
-	return nil, nil
+	span, _ := clh.trace("hotplugAddDevice")
+	defer span.Finish()
+
+	switch devType {
+	case vfioDev:
+		device := devInfo.(*config.VFIODev)
+
+		cl := clh.client()
+		ctx, cancel := context.WithTimeout(context.Background(), clhHotPlugAPITimeout*time.Second)
```

Comment:
We need to unify the way how we timeout multiple CLH API calls in a function. I chose to reuse the `clhAPITimeout` for each individual API call, and reset the timer after each API call with a call to `cancel()`. I feel this way is better, especially we don't need to add more hard-coded timeout. I am open to discuss about it

Reply:
Thanks for the comment yeah, I decide to add one more because I've seen that hotplug vfio device could take a bit more than an usual API all, I added a comment explaining  why is added